autogluon_zeroshot/metrics/_roc_auc_cpp/__init__.py

```diff
@@ -0,0 +1,38 @@
+import ctypes
+import os
+import numpy as np
+from numpy.ctypeslib import ndpointer
+
+
+class CppAuc:
+    """A python wrapper class for a C++ library, used to load it once and make fast calls after.
+    NB be aware of data types accepted, see method docstrings.
+    """
+
+    def __init__(self):
+        try:
+            self._handle = ctypes.CDLL(os.path.dirname(os.path.realpath(__file__)) + os.path.sep + "cpp_auc.so")
+        except OSError:
+            command_to_compile = f"cd {os.path.dirname(os.path.realpath(__file__))} && .{os.path.sep}compile.sh"
+            raise OSError(f'Missing cpp_auc.so compiled file... '
+                          f'You must first compile the C++ code to use this metric. '
+                          f'Run the below terminal command to compile the code:\n'
+                          f'{command_to_compile}')
```

Comment:
Given that we are just checking the file presence, it may be better to just to that directly to avoid catching errors not mentioned in the message:
```suggestion
        plugin_path = Path(__file__) / "cpp_auc.so"
        command_to_compile = f"cd {Path(__file__)} && .{os.path.sep}compile.sh"
        assert pluging_path.exists(), f'Missing cpp_auc.so compiled file... '
                          f'You must first compile the C++ code to use this metric. '
                          f'Run the below terminal command to compile the code:\n'
                          f'{command_to_compile}'

        self._handle = ctypes.CDLL(plugin_path)       
```

Reply:
Also, I was thinking that we could just run the compilation when the file is not found to avoid manual steps (those will complicate our setup if we want to run experiments remotely), I will send a PR to do that later.

Reply:
sounds good!